pkg/injector/patch_test.go

```diff
@@ -65,50 +146,44 @@ var _ = Describe("Test all patch operations", func() {
-			jsonPatches, err := wh.createPatch(&pod, req, proxyUUID)
+			pod := tests.NewPodFixture(namespace, podName, tests.BookstoreServiceAccountName, nil)
 
-			Expect(err).ToNot(HaveOccurred())
+			raw, err := json.Marshal(pod)
+			assert.NoError(err)
 
-			expectedJSONPatches := `[` +
-				// Add Volumes
-				`{"op":addOperation,` +
-				`"path":"/spec/volumes",` +
-				`"value":[{"name":"envoy-bootstrap-config-volume","secret":{"secretName":"envoy-bootstrap-config-proxy-uuid"}}]},` +
+			req := &admissionv1.AdmissionRequest{Namespace: namespace, Object: runtime.RawExtension{Raw: raw}}
+			rawPatches, err := wh.createPatch(&pod, req, proxyUUID)
 
-				// Add Init Container
-				`{"op":addOperation,` +
-				`"path":"/spec/initContainers",` +
-				`"value":[{"name":"osm-init","env":[{"name":"OSM_PROXY_UID","value":"1500"},` +
-				`{"name":"OSM_ENVOY_INBOUND_PORT","value":"15003"},{"name":"OSM_ENVOY_OUTBOUND_PORT","value":"15001"}],` +
-				`"resources":{},"securityContext":{"capabilities":{addOperation:["NET_ADMIN"]}}}]},` +
+			assert.NoError(err)
 
-				// Add Envoy Container
-				`{"op":addOperation,"path":"/spec/containers",` +
-				`"value":[{"name":"envoy","command":["envoy"],` +
-				`"args":["--log-level","","--config-path","/etc/envoy/bootstrap.yaml","--service-node","bookstore","--service-cluster","bookstore.-namespace-","--bootstrap-version 3"],` +
-				`"ports":[{"name":"proxy-admin","containerPort":15000},{"name":"proxy-inbound","containerPort":15003},{"name":"proxy-metrics","containerPort":15010}],` +
-				`"resources":{},"volumeMounts":[{"name":"envoy-bootstrap-config-volume","readOnly":true,"mountPath":"/etc/envoy"}],` +
-				`"imagePullPolicy":"Always",` +
-				`"securityContext":{"runAsUser":1500}}]},{"op":addOperation,"path":"/metadata/annotations",` +
-				`"value":{"prometheus.io/scrape":"true"}},` +
+			var patches []map[string]interface{}
+			err = json.Unmarshal(rawPatches, &patches)
+			assert.NoError(err)
 
-				// Add Prometheus Port Annotation
-				`{"op":addOperation,"path":"/metadata/annotations/prometheus.io~1port","value":"15010"},` +
+			var expectedPatches []map[string]interface{}
+			err = json.Unmarshal([]byte(tc.rawExpectedPatches), &expectedPatches)
+			assert.NoError(err)
 
-				// Add Prometheus Path Annotation
-				`{"op":addOperation,"path":"/metadata/annotations/prometheus.io~1path","value":"/stats/prometheus"},` +
+			assert.Len(patches, len(expectedPatches))
 
-				// Add Envoy UID Label
-				`{"op":replaceOperation,"path":"/metadata/labels/osm-proxy-uuid","value":"proxy-uuid"}` +
+		Outer:
+			// The patch order may change so assertions should be independent from it
+			for i, patch := range patches {
+				path := patch["path"].(string)
 
-				`]`
+				for _, expectedPatch := range expectedPatches {
+					expectedPatchPath := expectedPatch["path"].(string)
+					if path == expectedPatchPath {
+						assert.Equal(expectedPatch, patch)
+						continue Outer
```

Comment:
Just curious, why not just use `break` keyword? Or is this a preference?

Reply:
Yes, I actually mean why we use `continue Outer` instead of `break`. Although line label is supported, I personally don't find it widely used.